pkg/controllers/vdb/podfacts.go

```diff
@@ -566,8 +570,14 @@ func (p *PodFacts) checkForSimpleGatherStateMapping(_ context.Context, _ *vapi.V
 	pf.localDataSize = gs.LocalDataSize
 	pf.localDataAvail = gs.LocalDataAvail
 	pf.admintoolsExists = gs.AdmintoolsExists
-	// If the vertica process is running, then the database is UP. This is
-	// consistent with the liveness probe, which goes a bit further and checks
+	if vmeta.UseVClusterOps(vdb.Annotations) {
+		// In vclusterops mode, we call an HTTPS endpoint.
+		// If that returns HTTP code 200, then vertica is up
+		pf.upNode = gs.UpNode
+		return nil
```

Comment:
I suggest make the upnode check a function rather than exiting early here. If we add new checks in the future, we will need to work around this early exit. So, it could be prone to bugs.